code/modules/cooking/machinery/cooking_machines/container.dm

```diff
@@ -148,13 +144,119 @@
 	shortname = "shelf"
 	desc = "Put ingredients in this; designed for use with an oven. Warranty void if used."
 	icon_state = "ovendish"
-	center_of_mass = list("x" = 16,"y" = 12)
 	max_space = 30
-	max_reagents = 120
-	unacidable = TRUE
+	volume = 120
+	appliancetype = OVEN
+
+/obj/item/reagent_containers/cooking_container/skillet
+	name = "skillet"
+	shortname = "skillet"
+	desc = "Chuck ingredients in this to fry something on the stove."
+	icon_state = "skillet"
+	volume = 15
+	force = 15
+	hitsound = 'sound/weapons/smash.ogg'
+	flags = OPENCONTAINER // Will still react
+	appliancetype = SKILLET
+
+/obj/item/reagent_containers/cooking_container/skillet/Initialize(var/mat_key)
+	. = ..()
+	var/material/material = SSmaterials.get_material_by_name(mat_key || MATERIAL_STEEL)
+	if(material.name != MATERIAL_STEEL)
+		color = material.icon_colour
+	name = "[material.display_name] [initial(name)]"
+
+/obj/item/reagent_containers/cooking_container/saucepan
+	name = "saucepan"
+	shortname = "saucepan"
+	desc = "Is it a pot? Is it a pan? It's a saucepan!"
+	icon_state = "pan"
+	volume = 60
+	slot_flags = SLOT_HEAD
+	force = 15
+	hitsound = 'sound/weapons/smash.ogg'
+	flags = OPENCONTAINER // Will still react
+	appliancetype = SAUCEPAN
+
+/obj/item/reagent_containers/cooking_container/pan/Initialize(var/mat_key)
+	. = ..()
+	var/material/material = SSmaterials.get_material_by_name(mat_key || MATERIAL_STEEL)
+	if(material.name != MATERIAL_STEEL)
+		color = material.icon_colour
+	name = "[material.display_name] [initial(name)]"
+
+/obj/item/reagent_containers/cooking_container/pot
+	name = "cooking pot"
+	shortname = "pot"
+	desc = "Boil things with this. Maybe even stick 'em in a stew."
+	icon_state = "pot"
+	max_space = 50
+	volume = 180
+	force = 15
+	hitsound = 'sound/weapons/smash.ogg'
+	flags = OPENCONTAINER // Will still react
+	appliancetype = POT
+	w_class = ITEMSIZE_LARGE
+
+/obj/item/reagent_containers/cooking_container/pot/Initialize(var/mat_key)
+	. = ..()
+	var/material/material = SSmaterials.get_material_by_name(mat_key || MATERIAL_STEEL)
+	if(material.name != MATERIAL_STEEL)
+		color = material.icon_colour
+	name = "[material.display_name] [initial(name)]"
 
 /obj/item/reagent_containers/cooking_container/fryer
 	name = "fryer basket"
 	shortname = "basket"
 	desc = "Put ingredients in this; designed for use with a deep fryer. Warranty void if used."
-	icon_state = "basket"
\ No newline at end of file
+	icon_state = "basket"
+	appliancetype = FRYER
+
+/obj/item/reagent_containers/cooking_container/plate
+	name = "serving plate"
+	shortname = "plate"
+	desc = "A plate. You plate foods on this plate."
+	icon_state = "plate"
+	appliancetype = MIX
+	volume = 5 // for things like jelly sandwiches etc
+
+/obj/item/reagent_containers/cooking_container/plate/MouseDrop(var/obj/over_obj)
+	if(!istype(over_obj, /mob/living) || !Adjacent(over_obj))
```

Comment:
```suggestion
	if(!isliving(over_obj)) || !Adjacent(over_obj))
```
Also aren't these checks covered by use_check? You'll need to use it anyway here since you're not checking for unconsciousness